sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPSensor.java

```diff
@@ -112,27 +115,37 @@ public void describe(SensorDescriptor descriptor) {
   @Override
   public void execute(SensorContext context) {
     this.fileSystem = context.fileSystem();
+    this.context = context;
 
     FilePredicate mainFilePredicate = this.fileSystem.predicates().and(
       this.fileSystem.predicates().hasType(InputFile.Type.MAIN),
       this.fileSystem.predicates().hasLanguage(Php.KEY));
 
-    ImmutableList<PHPCheck> visitors = getVisitors(new CpdVisitor(context));
-
-    PHPAnalyzer phpAnalyzer = new PHPAnalyzer(fileSystem.encoding(), visitors);
-    ArrayList<InputFile> inputFiles = Lists.newArrayList(fileSystem.inputFiles(mainFilePredicate));
+    ImmutableList.Builder<PHPCheck> visitors = ImmutableList.<PHPCheck>builder().addAll(checks.all());
+    if (inSonarQube()) {
+      visitors.add(new CpdVisitor(context));
+    }
 
-    ProgressReport progressReport = new ProgressReport("Report about progress of PHP analyzer", TimeUnit.SECONDS.toMillis(10));
-    progressReport.start(Lists.newArrayList(fileSystem.files(mainFilePredicate)));
+    PHPAnalyzer phpAnalyzer = new PHPAnalyzer(fileSystem.encoding(), visitors.build());
+    Collection<CompatibleInputFile> inputFiles = CompatibilityHelper.wrap(fileSystem.inputFiles(mainFilePredicate), context);
 
-    Map<File, Integer> numberOLinesOfCode = new HashMap<>();
+    Map<String, Integer> numberOLinesOfCode = new HashMap<>();
 
-    analyseFiles(context, phpAnalyzer, inputFiles, progressReport, numberOLinesOfCode);
+    try {
+      analyseFiles(context, phpAnalyzer, inputFiles, numberOLinesOfCode);
+      if (inSonarQube()) {
+        processCoverage(numberOLinesOfCode);
+      }
+    } catch (CancellationException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)